src/zenml/models/v2/core/event_source.py

```diff
@@ -0,0 +1,158 @@
+
+# ------------------ Request Model ------------------
+
+
+class EventSourceRequest(WorkspaceScopedRequest):
+    """BaseModel for all event sources."""
+
+    name: str = Field(
+        title="The name of the stack component.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    flavor: str = Field(
+        title="The flavor of event source.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    plugin_type: PluginType = Field(
+        title="The plugin type of the event source.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    plugin_subtype: PluginSubType = Field(
+        title="The plugin subtype of the event source.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    description: str = Field(
+        default="",
+        title="The description of the event source.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+
+    configuration: Dict[str, Any] = Field(
+        title="The event source configuration.",
+    )
+
+
+# ------------------ Update Model ------------------
+
+
+@update_model
+class EventSourceUpdate(EventSourceRequest):
+    """Update model for event sources."""
+
+    name: str = Field(
+        title="The name of the stack component.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    description: str = Field(
+        default="",
+        title="The description of the event source.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    configuration: Dict[str, Any] = Field(
+        title="The event source configuration.",
+    )
+
+
+# ------------------ Response Model ------------------
+
+
+class EventSourceResponseBody(WorkspaceScopedResponseBody):
+    """ResponseBody for event sources."""
+
+    created: datetime = Field(
+        title="The timestamp when this event filter was created."
+    )
+    updated: datetime = Field(
+        title="The timestamp when this event filter was last updated.",
+    )
+
+
+class EventSourceResponseMetadata(WorkspaceScopedResponseMetadata):
+    """Response metadata for event sources."""
+
+    configuration: Dict[str, Any] = Field(
+        title="The event source configuration.",
+    )
+
+
+class EventSourceResponse(
+    WorkspaceScopedResponse[
+        EventSourceResponseBody, EventSourceResponseMetadata
+    ]
+):
+    """Response model for event sources."""
+
+    name: str = Field(
+        title="The name of the event source.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    flavor: str = Field(
+        title="The flavor of event source.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    plugin_type: PluginType = Field(
+        title="The plugin type of the event source.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    plugin_subtype: PluginSubType = Field(
+        title="The plugin subtype of the event source.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    description: str = Field(
+        default="",
+        title="The description of the event source.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
```

Comment:
If I understand correctly how the body/metadata separation works, some of these need to go in the body while others need to go in the metadata and there need to be properties defined here referencing one or the other

Reply:
yeah, i actually have struggled with this and decided to just do it one way and look at it again. Also we've had discussions around the API design and one of the proposals was basically all properties go in the main response, the body is filled with light relationship references and the metadata is for heavy stuff that is not needed in lists. But all of this is still up in the air @bcdurak whats the status?